File: Escola/Cursos.Api/scripts/service/EventoturnosService.js
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/13/2017 1:58:38 PM
///// Gerado automaticamente para alteracoes dos metodos comum alterar o template
///// </sumary>
"use strict";
app.factory("EventoturnosService", ["$http", "ngAuthSettings", function ($http, ngAuthSettings) {
        var service = {};
		var serviceBase = ngAuthSettings.apiServiceBaseUri;
        service.GetAll = function (callback) {
            $http.get(serviceBase + "Eventoturnos/getall")
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.GetById = function (id, callback) {
            $http.get(serviceBase + "Eventoturnos/getbyid/" + id)
               .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Post = function (Eventoturnos, callback) {
            $http.post(serviceBase + "Eventoturnos/post", Eventoturnos)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        service.Put = function (Eventoturnos, callback) {
            $http.post(serviceBase + "Eventoturnos/put", Eventoturnos)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

		service.Delete = function (Eventoturnos, callback) {
            $http.post(serviceBase + "Eventoturnos/delete", Eventoturnos)
                .then(function (response) {
                    callback(response);
                }, function (response) {
                    callback(response);
                });
        };

        return service;
    }]);

File: Escola/Cursos.Repository/Services/EventoturnosService.cs
///// <sumary>
///// By: Jeferson Braga do Carmo Create Date: 25/11/2015
///// Last Update: 11/11/2017 9:23:24 PM

///// Para outros metodos implementar classe partial
///// </sumary>
using System;
using System.Linq;
using System.Collections.Generic;
using System.Threading.Tasks;
using Cursos.Repository.Services.Base;
using Cursos.Data;
using Cursos.Repository.Services.Interfaces;
using System.Linq.Expressions;
using Cursos.Repository.Work.Interface;
namespace  Cursos.Repository.Services
{
	  /// <summary>
      /// Public Service Class  EventoturnosService
      /// </summary>
      public partial class EventoturnosService : BaseServices, IEventoturnosService
      {

		public EventoturnosService(IUnitOfWork unitOfWork)
        {
			UnitOfWork = unitOfWork;
        }

		/// <summary>
        /// Add Range Entity EV_EVENTOTURNOS
        /// </summary>
        /// <param name="entity"></param>
		public void Add(IEnumerable<EV_EVENTOTURNOS> entity)
        {
            UnitOfWork.EventoturnosRepository.Add(entity);
        }

		/// <summary>
        /// Add Entity EV_EVENTOTURNOS
        /// </summary>
        /// <param name="entity"></param>
		public void Add(EV_EVENTOTURNOS entity)
        {	
            UnitOfWork.EventoturnosRepository.Add(entity);
        }
		/// <summary>
        /// Delete Entity EV_EVENTOTURNOS
        /// </summary>
        /// <param name="entity"></param>
        public void Delete(EV_EVENTOTURNOS entity)
        {
            UnitOfWork.EventoturnosRepository.Delete(entity);
        }
		 /// <summary>
        /// Get All Entity EV_EVENTOTURNOS
        /// </summary>
        /// <returns>IEnumerable EV_EVENTOTURNOS</returns>
        public IEnumerable<EV_EVENTOTURNOS> GetAll()
        {
			return UnitOfWork.EventoturnosRepository.GetAll();
        }
	     
		/// <summary>
        /// Update Entity EV_EVENTOTURNOS
        /// </summary>
        /// <param name="entity"></param>
        public void Update(EV_EVENTOTURNOS entity)
        {
            UnitOfWork.EventoturnosRepository.Edit(entity);
        }

		/// <summary>
        /// Commit Transaction
        /// </summary>
        /// <returns></returns>
        public int Commit()
        {
            return UnitOfWork.Commit();
        }
		/// <summary>
        /// Commit Async Transaction
        /// </summary>
		public Task<int> CommitAsync()
        {
            return UnitOfWork.CommitAsync();
        }
				/// <summary>
        /// Get Entity EV_EVENTOTURNOS By Id
        /// </summary>
        /// <param name="id"></param>
        /// <returns>EV_EVENTOTURNOS</returns>
		public EV_EVENTOTURNOS GetById(int id)
        {
			return UnitOfWork.EventoturnosRepository.FindBy(x => x.IDEVENTO.Equals(id)).SingleOrDefault();
        }		
				
		public IEnumerable<EV_EVENTOTURNOS> FindBy(Expression<Func<EV_EVENTOTURNOS, bool>> predicate)
        {
            return UnitOfWork.EventoturnosRepository.FindBy(predicate);
        }
      }
}
